oak/server/rust/oak_runtime/src/runtime/introspect.rs

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use hyper::{
+    service::{make_service_fn, service_fn},
+    Body, Error, Method, Request, Response, Server, StatusCode,
+};
+use log::info;
+use regex::Regex;
+use std::{net::SocketAddr, sync::Arc};
+
+use crate::runtime::Runtime;
+
+// TODO(#672): Shift to a templating library.
+fn dot_wrap(title: &str, graph: &str) -> String {
+    format!(
+        r###"
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>{}</title>
+<body>
+<script src="//d3js.org/d3.v5.min.js"></script>
+<script src="https://unpkg.com/@hpcc-js/wasm@0.3.6/dist/index.min.js"></script>
+<script src="https://unpkg.com/d3-graphviz@3.0.0/build/d3-graphviz.js"></script>
```

Comment:
Since you are at it, could you add subresource integrity fields to these? Effectively the hashes of the exact versions that we depend on. See https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity

Reply:
Done.